### ISSUE TITLE
The `jvmrunargs` lookup is always empty

### ISSUE THREAD

**Issue by ppkarwasz** [apache/logging-log4j2#2726]
The [`JmxRuntimeInputArgumentsLookup`](org.apache.logging.log4j.core.lookup.JmxRuntimeInputArgumentsLookup) lookup created with the standard constructor always returns `null`.

The lookup has a singleton instance `JMX_SINGLETON`, which is not used anywhere in the code.

**Comment by ppkarwasz** [comment]
Sure, go ahead! Thanks for picking this up!

Out of curiosity, are you using the `jvmrunargs` lookup yourself, or just contributing to improve things? This lookup may have **limited** usefulness in practice, since many applications already parse JVM or CLI arguments and store them in system properties or an environment-like structure.

For example, in Spring Boot, CLI options of the form `--<key>=<value>` are automatically placed in the application environment, so this lookup wouldn’t add much value there.

**Comment by ramanathan1504** [comment]
Thanks for the quick response!

I originally noticed this while reviewing the available lookups in Log4j2 and got curious about ${jvmrunargs} — mainly for completeness and tooling use. You’re right that in modern frameworks like Spring Boot, most CLI or JVM args get surfaced via environment abstractions or system properties, so this lookup might not add much there.

That said, I’ve worked on some legacy projects (including older Java Applet-based systems) where we had to tweak and inspect JVM-level arguments directly — especially for things like agent loading, memory flags, or debug switches. In those scenarios, being able to dynamically log or inspect JVM run args at runtime was helpful for monitoring and debugging.

I’m happy to wire up the missing JMX_SINGLETON so the lookup returns expected values, or alternatively help document its current limitations. Let me know which direction you’d prefer — I can open a PR if this seems worth fixing.

**Comment by ppkarwasz** [comment]
Thanks for you explanation and offer to help!

Sure, as PR would be welcome!

**Comment by vy** [comment]
This plugin

1. has never been documented, and
2. has never worked correctly.

Am I correct?

**Comment by ramanathan1504** [comment]
You’re correct that previously this lookup was undocumented and didn’t function as intended.  

This PR fixes that by:
- Registering the `jvmrunargs` lookup in the `Interpolator`, enabling retrieval of JVM runtime arguments via `${jvmrunargs:...}` expressions.
- Adding `JvmRunArgsLookupTest` and `JvmRunArgsIntegrationTest` (with a sample `log4j2-test.xml`) to verify correct registration and behavior in both null and non-null cases.
- Manually verifying the lookup now returns JVM runtime arguments:

```java
StrLookup lookup = new Interpolator();
System.out.println(lookup.lookup("jvmrunargs:0")); // e.g. "-DexampleArg=value"

**Comment by ramanathan1504** [comment]
If the team feels this feature is unnecessary, I’m happy to update the PR to remove it or adjust it and document accordingly.

**Comment by ppkarwasz** [comment]
Hi @vy,

> 1. has never been documented, and

It actually **was** documented on the old website:

https://github.com/apache/logging-log4j2/blob/fea2a7116160fb1555d578406444b4fc4f0ef2da/src/site/xdoc/manual/lookups.xml#L287-L301

I then deliberately **un**-documented it, because it didn't work (see #2753).

> 2. has never worked correctly.

Yes, from what I can tell it was born broken and stayed that way until… #3071. I believe I accidentally fixed it while making JMX optional (for the benefit of Android users), and then promptly forgot it even existed.

@ramanathan1504: can you check if this still reproduces on `2.25.0` or later? I think it’s already fixed, but remains still undocumented. If it is working, could you document it in the `lookups.adoc` page via PR?

**Comment by ramanathan1504** [comment]
Hi @ppkarwasz,

I’ve confirmed that the `jvmrunargs` lookup works in `2.25.0` and later, so I’ve added its documentation to `lookups.adoc` along with the existing changes. This PR now includes both the tests (unit + integration) and the docs in one place.

**Comment by ramanathan1504** [comment]
Understood. Do you want me to revert the entire PR content and just keep a changelog entry for the removal of `jvmrunargs`?

**Comment by ramanathan1504** [comment]
Updated the change log and revert old one in ssame pr #3874 

@vy 
I can’t access the ASF Slack link — would you mind summarizing the key points here?

**Comment by vy** [comment]
@ramanathan1504, the conversation is only accessible by maintainers, I've shared it for the records, you can ignore it. See my comments in #3874.
